spangres/toddl_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spangres
+
+import (
+	"testing"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/stretchr/testify/assert"
+)
+
+// This is just a very basic smoke-test for toSpannerType.
+// The real testing of toSpannerType happens in process_test.go
+// via the public API ProcessPgDump (see TestProcessPgDump).
+func TestToSpannerType(t *testing.T) {
+	conv := internal.MakeConv()
+	conv.SetSchemaMode()
+	name := "test"
+	srcSchema := schema.Table{
+		Name:     name,
+		ColNames: []string{"a", "b", "c", "d", "e", "f"},
+		ColDefs: map[string]schema.Column{
+			"a": schema.Column{Name: "a", Type: schema.Type{Name: "int8"}},
+			"b": schema.Column{Name: "b", Type: schema.Type{Name: "float4"}},
+			"c": schema.Column{Name: "c", Type: schema.Type{Name: "bool"}},
+			"d": schema.Column{Name: "d", Type: schema.Type{Name: "varchar", Mods: []int64{6}}},
+			"e": schema.Column{Name: "e", Type: schema.Type{Name: "numeric"}},
```

Comment:
Should we also add `date` type to unit test?

Reply:
toddl_test.go just has a very basic test of toddl.go.
The main test of toddl.go is in pgdump_test.go and infoschema_test.go.

There are a couple of options:
a) Add all missing cases to toddl_test.go e.g. both array, date. That would mean we test the Spangres changes, but it would also mean we don't test the bulk of spangres/toddl.go (which is the mostly the same code as posgres/toddl.go).
b) Add pgdump_test.go and infoschema_test.go to this directory, which would give us coverage, but is very messy.

I think (a) is the sane option, but we need to deal with the refactor of toddl.go soon.